File: Systems-for-Design-and-Implementation/Labs/Lab9/ProiectJava/TransportServices/src/main/java/transport/services/ITransportServices.java
package transport.services;

import transport.model.ReservationManager;
import transport.model.Seat;
import transport.model.Trip;
import transport.model.Reservation;

import java.util.List;

public interface ITransportServices {
    ReservationManager login(ReservationManager reservationManager, ITransportObserver client) throws TransportException;

    ReservationManager searchReservationManagerByName(String name) throws TransportException;

    Trip[] getAllTrips() throws TransportException;

    Trip searchTripById(Long id) throws TransportException;

    Reservation addReservation(Reservation reservation) throws TransportException;

    List<Reservation> getReservationsByTrip(Trip trip) throws TransportException;

    List<Seat> getSeatsByReservation(Long reservationId) throws TransportException;

    List<Seat> getSeatsByTrip(Trip trip) throws TransportException;

    ReservationManager logout(ReservationManager reservationManager, ITransportObserver client) throws TransportException;
}

File: S4-Systems-for-Design-and-Impl/Labs/Lab9/ProiectCSharp/TransportServices/transport.services/ITransportServices.cs
using System.Collections.Generic;
using Transport.Model.transport.model;
using TransportModel.transport.model;

namespace TransportServices.transport.services
{
    public interface ITransportServices
    {
        ReservationManager Login(ReservationManager reservationManager, ITransportObserver client);

        ReservationManager SearchReservationManagerByName(string name);

        Trip[] GetAllTrips();

        Trip SearchTripById(long id);

        Reservation AddReservation(Reservation reservation);

        List<Reservation> GetReservationsByTrip(Trip trip);

        List<Seat> GetSeatsByReservation(long reservationId);

        List<Seat> GetSeatsByTrip(Trip trip);

        ReservationManager Logout(ReservationManager reservationManager, ITransportObserver client);
    }
}